src/App.js

```diff
@@ -1,27 +1,76 @@
-import React from 'react'
+import React, { useState } from 'react'
 import Nav from './Nav'
 import Homepage from './Homepage'
 import Login from './Login'
 import About from './About'
 
 import { Route, Switch } from 'react-router-dom';
+import { gql, useQuery, useMutation } from '@apollo/client';
+
 const App = () => {
 
+const [user, setUser] = useState({})
+
+const [username, setUsername] = useState('')
+const [email, setEmail] = useState('')
+
+
+const CREATE_USER = gql`
+
+mutation createUser(
+  $username: String!
+  $email: String!
+  ) {
+  createUser( input: {
+    username: $username
+    email: $email
+  } ) {
+    user {
+    id
+    username
+    email
+    }
+  }
+}
+
+`
+
+const [createUser, { data, error, loading }] = useMutation(CREATE_USER)
+
+const loginUser = (username, email) => {
+  setUsername(username)
+  setEmail(email)
+
+  createUser({
+    variables: {
+      username: username,
+      email: email
+    }
+  })
+  .then((data) => setUser(data.data.createUser.user))
+
+
+  if (error) {
+    console.log(error)
```

Comment:
so because of the scope of this app you not need to worry about this... however, always caution sending a console.log to Javascript and the browser. It's usually best to always catch and raise errors on the backend and then log them to the Rails console. doing a `console.log` on the frontend might expose something to the user that you don't want them to see.

Reply:
Okay thats good a recommendation. We will work on changing it around the catch the error.

Reply:
That's a great point! We will change this in a future PR.

Reply:
You are right. We are going to do error handling definitely in the future!